ppr-ui/src/components/results/Results.vue

```diff
@@ -0,0 +1,180 @@
+
+export default defineComponent({
+  props: {
+    defaultHeaders: {
+      type: Array as () => TableHeadersIF
+    },
+    defaultResults: {
+      type: Array as () => Array<SearchResultIF>
+    },
+    defaultSelected: {
+      type: Array as () => Array<SearchResultIF>
+    },
+    resultsGetter: {
+      type: String,
+      default: 'getSearchResults'
+    }
+  },
+  setup (props) {
+    const { getSearchResults } = useGetters<any>([props.resultsGetter])
+    const localState = reactive({
+      searched: false,
+      searchValue: '',
+      selected: props.defaultSelected,
+      headers: props.defaultHeaders,
+      results: props.defaultResults,
+      exactMatchesLength: 0,
+      totalResultsLength: 0,
+      selectedLength: computed((): number => {
+        return localState.selected?.length | 0
+      }),
+      searchResponse: computed((): SearchResponseIF => {
+        const resp = getSearchResults.value
+        if (resp) {
+          localState.searchValue = resp.searchQuery.criteria.value
+          localState.searched = true
+          localState.headers = tableHeaders[resp.searchQuery.type]
+          localState.results = resp.results
+          localState.totalResultsLength = resp.totalResultsSize
+          return resp
+        }
+        return {
+          searchId: '',
+          maxResultsSize: NaN,
+          totalResultsSize: NaN,
+          returnedResultsSize: NaN,
+          searchQuery: {
+            type: '',
+            criteria: {}
+          },
+          results: []
+        }
+      })
+    })
+    const getClass = (item:SearchResultIF):string => {
+      if (item.matchType === MatchTypes.EXACT) {
+        return 'exact-match'
+      }
+      return 'normal-match'
+    }
+    watch(() => localState.results, (results) => {
+      const selectedExactMatches = []
+      let count = 0
+      let x:any
+      for (x in results) {
+        if (results[x].matchType === MatchTypes.EXACT) {
+          count += 1
+          selectedExactMatches.push(results[x])
+        }
+      }
+      localState.exactMatchesLength = count
+      localState.selected = selectedExactMatches
+    })
+
+    return {
+      ...toRefs(localState),
+      getClass
+    }
+  }
+})
+</script>
+
+<style lang="scss">
+@import '@/assets/styles/theme.scss';
+#search-btn {
+  width: 8rem;
+}
+#search-btn-info {
+  color: $gray8;
+  font-size: 0.725rem;
+}
+.close-popup-btn {
+  background-color: transparent !important;
+}
+.exact-match td {
+  color: $gray9 !important;
+  font-weight: bold;
+}
+.normal-match td {
+  color: $gray9 !important;
+}
+.main-results-div {
+  width: 100%;
+}
+.result-info {
+  color: $gray9 !important;
+  font-size: 0.825rem;
+}
+.no-results-info {
+  color: $gray9 !important;
+  font-size: 0.825rem;
+  width: 100%;
+  height: 8rem;
+}
+// .result-header {
+//   font-size: 0.825rem !important;
+//   color: $gray7 !important;
+// }
+// .results-table {
+//   height: 500px;
+// }
+.v-select-list {
```

Comment:
Potentially some unused classes/id's that could removed if they are not used within the template.

Reply:
yes definitely